File: LeetCode/Pascal's Triangle.py
class Solution:
	# @return a list of lists of integers
	def generate(self, numRows):
		ans=[]
		if numRows == 0:
			return ans
		ans = [[1]]
		l = 2
		for i in range(1, numRows):
			cursor = 0
			ans.append([])
			for j in range(0, l):
				if j == 0:
					t = ans[i-1][0]
				elif j == l - 1:
					t = ans[i-1][l-2]
				else:
					t = ans[i-1][j] + ans[i-1][j-1]
				ans[i].append(t)
			l += 1
		return ans

if __name__ == '__main__':
	s = Solution()
	print s.generate(10)

File: LeetCode/Pascal's Triangle.cpp
class Solution {
public:
    vector<vector<int> > generate(int numRows) {
        vector<vector<int> > triangles;
        if (numRows == 0) return triangles;
        triangles.push_back(vector<int>(1,1));
        for (int i = 1; i < numRows; ++i){
            vector<int> tmp;
            tmp.push_back(triangles[i-1][0]);
            for (int j = 1; j < triangles[i-1].size(); ++j)
                tmp.push_back(triangles[i-1][j-1]+triangles[i-1][j]);
            tmp.push_back(triangles[i-1][triangles[i-1].size()-1]);
            triangles.push_back(tmp);
        }
        return triangles;
    }
};